tonilopezmr/day5/MazeOfJumpInstructionsPartTwo.kt

```diff
@@ -0,0 +1,64 @@
+package day5
+
+import org.junit.Assert.*
+import org.junit.Test
+import java.io.File
+import java.util.*
+
+class MazeOfJumpInstructionsPartTwo {
+
+  private fun jumpInstI(input: List<Int>): Int {
+    var offset = 0
+    var steps = 0
+    val instructions = input.toMutableList()
+
+    while (offset < input.size) {
+      val aux = instructions[offset]
+      instructions.removeAt(offset)
+      instructions.add(offset, if (aux >= 3) aux - 1 else aux + 1)
+      offset += aux
+      steps++
+    }
+
+    return steps
+  }
+
+  private fun jumpInst(input: List<Int>): Int = jumpInst(0, 0, input)
+
+  tailrec private fun jumpInst(offset: Int, step: Int, input: List<Int>): Int {
+    return if (offset >= input.size) step
+    else jumpInst(offset + input[offset], step + 1, input.take(offset) + newOffset(input, offset) + input.takeLast(input.size - (offset + 1)))
+  }
+
+  private fun newOffset(input: List<Int>, offset: Int) = if (input[offset] >= 3) input[offset] - 1 else input[offset] + 1
```

Comment:
By passing this function as a parameter you could have reutilized the same class than for part 1 😸

Reply:
yes !